rcall.c

```diff
@@ -764,10 +781,21 @@ static void pg_get_one_r(char *value,  plcDatatype column_type, SEXP *obj, int e
                            plc_get_type_name(column_type), column_type);
             break;
 
+        case PLC_DATA_BYTEA:
         case PLC_DATA_TEXT:
         default:
             /* Everything else is defaulted to string */
-            SET_STRING_ELT(*obj, elnum, COPY_TO_USER_STRING(((char *)value)));
+			if (strlen(value) != 0 && column_type == PLC_DATA_BYTEA){
```

Comment:
Move the code for PLC_DATA_BYTEA to the PLC_DATA_BYTEA branch.
Use (value[0] != '\0') instead.